modules/govuk_jenkins/manifests/init.pp

```diff
@@ -6,51 +6,66 @@
 # === Parameters:
 #
 # [*apt_mirror_hostname*]
-#   Hostname to use for the APT mirror.
+#   Hostname of the apt mirror
 #
 # [*github_enterprise_cert*]
 #   PEM certificate for GitHub Enterprise.
 #
+# [*github_enterprise_hostname*]
+#   The hostname of Github Enterprise
+#
+# [*jenkins_home*]
+#   Home directory of the Jenkins user
+#
+# [*jenkins_user*]
+#   The username of the Jenkins user
+#
 # [*config*]
 #   A hash of Jenkins config options to set
 #
 # [*plugins*]
 #   A hash of Jenkins plugins that should be installed
 #
+# [*version*]
+#   The version of Jenkins to install
+#
+# [*ssh_private_key*]
+#   The SSH private key of the Jenkins user
+#
+# [*ssh_public_key*]
+#   The SSH public key of the Jenkins user
+#
 class govuk_jenkins (
   $apt_mirror_hostname,
   $github_enterprise_cert,
   $github_enterprise_hostname,
-  $jenkins_home,
+  $jenkins_home = '/var/lib/jenkins',
+  $jenkins_user = 'jenkins',
   $github_enterprise_cert_path,
   $config = {},
   $plugins = {},
+  $version = '1.554.2',
+  $ssh_private_key = undef,
+  $ssh_public_key = undef,
 ) {
   validate_hash($config, $plugins)
 
-  include govuk_python
-  include govuk_jenkins::job_builder
-  include govuk_jenkins::ssh_key
-  include govuk_jenkins::config
+  include ::govuk_python
+  include ::govuk_jenkins::config
+  include ::govuk_jenkins::job_builder
 
-  user { 'jenkins':
-    ensure     => present,
-    home       => $jenkins_home,
-    managehome => true,
-    shell      => '/bin/bash',
+  class { 'govuk_jenkins::user':
+    home_directory => $jenkins_home,
+    username       => $jenkins_user,
+    private_key    => $ssh_private_key,
+    public_key     => $ssh_public_key,
   }
 
-  include govuk_java::openjdk7::jdk
-  include govuk_java::openjdk7::jre
-
-  class { 'govuk_java::set_defaults':
-    jdk     => 'openjdk7',
-    jre     => 'openjdk7',
-    require => [
-                  Class['govuk_java::openjdk7::jdk'],
-                  Class['govuk_java::openjdk7::jre'],
-                ],
-    notify  => Class['jenkins::service'],
+  class { 'govuk_jenkins::package':
+    apt_mirror_hostname => $apt_mirror_hostname,
+    config              => $config,
+    plugins             => $plugins,
+    version             => $version,
```

Comment:
Instead of settings this on `govuk_jenkins` and passing it through to `govuk_jenkins::package` we can set this in hiera as `govuk_jenkins::package::version` and avoid having the default in 2 places.